presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSourceBase.java

```diff
@@ -187,11 +188,21 @@ else if (type instanceof DateType) {
         }
     }
 
-    private long parseTimestamp(String value)
+    @VisibleForTesting
+    static long parseTimestamp(String value)
     {
         try {
             return parseLong(value);
         }
+        catch (NumberFormatException e) {
+            // Ignore here since this could be in a tight loop
+        }
+        // Sometimes Pinot returns float point string in a field that we expect timestamp.
+        // This can happen because of JsonParser, Pinot Query Engine, etc.
+        // In this case we may still go through by reading a float value
+        try {
+            return parseDouble(value).longValue();
+        }
         catch (Exception e) {
             return TimestampUtils.toMillisSinceEpoch(value);
         }
```

Comment:
- Lets move this method to `PinotUtils.java` as it is a generic util method 
- make it a nested `try-catch`. Like move the second `try-catch` to inside the first `catch`

Reply:
I don't have a strong opinion either should be fine.